OHCRN/platform
# 💄 #355: reset notifications on page change

## Description of Changes

### Consent UI
- Add [router events handler](https://nextjs.org/docs/app/api-reference/functions/use-router#router-events)
- Clear notification context on page change
- Testing:
  - Consent completion notification
    - Inspect the NotificationsProvider in React DevTools.
    - Complete step 5 of Consent Wizard, which takes you to the Dashboard.
    - A green notification should be visible.
    - Click on the OHCRN logo at the top left of the page to visit the homepage.
    - Notification is removed from the context provider.

## PR Readiness Checklist

- [x] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Builds locally without errors or warnings
- [x] Tests are updated (if required) and passing
- [x] PR feedback has been addressed **if applicable**
- [x] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files
